articles/howtos/create-new-decorator.md

```diff
@@ -0,0 +1,222 @@
+
+```csharp
+public string ConfigureCode()
+{
+    return @"app.UseStaticFiles(new StaticFileOptions
+    {
+        FileProvider = new PhysicalFileProvider(
+            Path.Combine(Directory.GetCurrentDirectory(), ""MyStaticFiles"")),
+        RequestPath = ""/StaticFiles""
+    });";
+}
+```
+
+Also, with the Declare Usings section, we need to specify some namespaces that our Decorator will need to successfully apply code that will compile.
+
+Override the `DeclareUsings` method to look like this:
+
+```csharp
+[IntentManaged(Mode.Merge, Body = Mode.Ignore, Signature = Mode.Fully)]
+public IEnumerable<string> DeclareUsings()
+{
+    return new string[]
+    {
+        "System.IO",
+        "Microsoft.Extensions.FileProviders"
+    };
+}
+```
+
+We now need to add a dependency on the original Template.
+
+Open up the `MyCompany.MyDecoratorModule.imodspec` file in Visual Studio.
+
+You will see something resembling this:
+
+```xml
+<?xml version="1.0" encoding="utf-8"?>
+<package>
+  <id>MyCompany.MyDecoratorModule</id>
+  <version>1.0.0</version>
+  <summary>A custom module for MyCompany.</summary>
+  <description>A custom module for MyCompany.</description>
+  <authors>MyCompany</authors>
+  <templates></templates>
+  <dependencies>
+    <dependency id="Intent.Common" version="2.0.0" />
+    <dependency id="Intent.Common.Types" version="2.0.0" />
+  </dependencies>
+  <files>
+    <file src="$outDir$/$id$.dll" />
+    <file src="$outDir$/$id$.pdb" />
+  </files>
+  <decorators>
+    <decorator id="MyDecoratorModule.StaticFileServerDecorator" />
+  </decorators>
+</package>
+```
+
+In the `dependencies` section we need to add a dependency to our original Template Module.
+
+```xml
+<dependencies>
+    <dependency id="Intent.Common" version="2.0.0" />
+    <dependency id="Intent.Common.Types" version="2.0.0" />
+    <dependency id="MyCompany.MyModule" version="1.0.0" />
+</dependencies>
+```
+
+Where did we get this info?
+Open up the `MyCompany.MyModule.imodspec` file in the project we imported.
+
+At the top of that XML file you will see this:
+
+```xml
+<?xml version="1.0" encoding="utf-8"?>
+<package>
+  <id>MyCompany.MyModule</id>
+  <version>1.0.0</version>
+```
+
+We only need the `id` and `version` fields and that gets used to specify module dependencies.
+
+Now I can compile the solution.
+
+## Install the Decorator Module in your application
+
+Open up the `Test.App` in Intent Architect, click on the `Modules` on the side panel.
+
+Click on the "gear" icon located to the right. Add the location to the new Module.
+
+![Asset Repo](images/create-new-decorator/AssetRepo.png)
+
+**Name**: MyDecoratorModule
+**Address**: ./MyDecoratorModule/Intent.Modules
+
+Now, select the `MyDecoratorModule` in the dropdown.
+
+![Ready To Install My Decorator](images/create-new-decorator/ReadyToInstallMyDecorator.png)
+
+
+Locate the `MyCompany.MyDecoratorModule` and click on the `Install` button (located to the right).
+
```

Comment:
Again, I'm wondering that if we make this how-to assume the previous how-to as a starting point, then we could stream line this how-to and largely avoid this section. Ideally this should be possible to also set up in the Module Builder.

It also presents an Intent Architect feature idea where the module builder should be able to capture these dependencies, if it can't already.

Reply:
I think we can have the asset repo docs defined in a separate page and just have a small reference in the how-to. But to imply that the previous starting point is the way to go, is a problem. Especially if you take out the link in the beginning of the page which states that. But even if you don't, you certainly don't want to convoluted the previous how-to with concerns that it is not supposed to be concerned with.

Reply:
Building on my [big comment](https://github.com/IntentSoftware/Docs/pull/1/files#r366506549), for the sake of keeping the article concise, let's assume they probably:
- ...know about module dependencies, but it's a good idea to put a HINT which links to an article which explains them.
- ...know about adding repositories to Intent and installing modules, so we just need to tell the reader the module needs to be installed, but it's a good idea to have a link on how to do so.

```suggestion
In your Intent Architect application, [install or re-install the module](xref:TODO).

> [!HINT]
> (Optional.) You can refer to the [](xref:TODO) article if you would like this module to depend upon the module with the template interface and automatically install it too.
```